server/src/services/search.service.ts

```diff
@@ -113,14 +113,25 @@ export class SearchService extends BaseService {
     }
 
     const userIds = this.getUserIdsToSearch(auth);
-    const key = machineLearning.clip.modelName + dto.query + dto.language;
-    let embedding = this.embeddingCache.get(key);
-    if (!embedding) {
-      embedding = await this.machineLearningRepository.encodeText(machineLearning.urls, dto.query, {
-        modelName: machineLearning.clip.modelName,
-        language: dto.language,
-      });
-      this.embeddingCache.set(key, embedding);
+    let embedding;
+    if (dto.query) {
```

Comment:
This validation should move to the dto. You can throw an error if one of the two fields are not provided there. Then here, you can just do something like:
```ts
let embedding;
if(dto.exampleAssetId) {
  embedding = ...
}
  
  
if(dto.query) {
  embedding = ...
}
 ```